scripts/generate-locales.ts

```diff
@@ -0,0 +1,118 @@
+import ts from 'typescript';
+import path, { relative } from 'path';
+import { writeFile, mkdir, stat } from 'fs/promises';
+import { glob, chalk } from 'zx';
+
+const SRC_DIR = path.resolve('./src/lib/i18n/locales');
+const DEST_DIR = path.resolve('./static/i18n/locales');
+
+const logAndExit = (msg: string, logError: (msg: string) => void) => {
+  logError(chalk.red(msg));
+
+  if (logError === console.error) {
+    process.exit(0);
+  }
+};
+
+const relativeTo = (path: string) => {
+  return relative('./', path);
+};
+
+export const generateLocales = async (
+  src: string = SRC_DIR,
+  dest: string = DEST_DIR,
+  logError: (msg: string) => void = console.error,
+) => {
+  const pattern = path.join(src, '*/*.ts');
+  const filesToTranslate = await glob(pattern);
+
+  filesToTranslate.map(async (source) => {
+    const program = ts.createProgram([source], ts.getDefaultCompilerOptions());
+    const sourceFile = program.getSourceFile(source);
+    const checker = program.getTypeChecker();
+
+    if (!sourceFile) {
+      logAndExit(
+        `Error retrieving source file information for ${source}`,
+        logError,
+      );
+    }
+
+    const sourceFileSymbol = checker.getSymbolAtLocation(sourceFile!);
+
+    if (!sourceFileSymbol) {
+      logAndExit(`Error parsing source file: ${source}`, logError);
+    }
+
+    const defaultExport = checker.tryGetMemberInModuleExports(
+      'default',
+      sourceFileSymbol!,
+    );
+
+    if (!defaultExport) {
+      logAndExit(`No default export in source: ${source}`, logError);
+    }
+
+    const jsObject: Record<string, string> = {};
```

Comment:
I hate the name of this object, but naming stuff is hard.

Reply:
I can do better 😂